onnxruntime/core/mlas/inc/mlas_qnbit.h

```diff
@@ -77,3 +77,134 @@ MlasIsSQNBitGemmAvailable(
     size_t BlkBitWidth,
     size_t BlkLen
 );
+
+/**
+ * @brief Define compute types of block quantization
+ */
+typedef enum {
+    CompUndef = 0, /*!< undef */
+    CompFp32 = 1,  /*!< input fp32, accumulator fp32 */
+    CompFp16 = 2,  /*!< input fp16, accumulator fp16 */
+    CompBf16 = 3,  /*!< input bf16, accumulator fp32 */
+    CompInt8 = 4   /*!< input int8, accumulator int32 */
+} MLAS_COMPUTE_TYPE;
+
+/**
+ * @brief Data parameters for Q4 GEMM routine
+ *        C = A * B + Bias
+ *        A must be a float32 matrix
+ *        B must be a quantized and packed int4 blob
+ *        All except C are [in] parameters
+ */
+struct MLAS_NBITS_GEMM_DATA_SIMPLE_PARAMS {
+    const float* A = nullptr; /**< address of A (float32 matrix)*/
+    const void* B = nullptr;  /**< address of B (quantized and packed int4 blob)*/
+    float* C = nullptr;       /**< address of result matrix */
+    size_t lda = 0;           /**< leading dimension of A */
+    size_t ldc = 0;           /**< leading dimension of C*/
+};
+
+/**
+ * @brief Check if the parameter combination is supported by the runtime device.
+ *
+ * @param N      the number of columns of matrix B.
+ * @param K      the number of rows of matrix B.
+ * @param block_size    size of the block to quantize, elements from the same block share the same
+ * scale and zero point
+ * @param nbits  number of bits used for weight quantization (default 4)
+ * @param is_asym  flag for asymmetric quantization
+ * @param comp_type  specify input data type and accumulator data type
+ * @return support flag, true if the combination is supported.
+ */
+bool MLASCALL
+MlasIsNBitGemmAvailable(size_t N, size_t K, size_t block_size, int nbits, bool is_asym, MLAS_COMPUTE_TYPE comp_type);
```

Comment:
Could we just keep the MlasNBitsGemmPackBSize function and use it to determine if we should leverage jblas and prepack the weight?

Reply:
Done, use MlasNBitsGemmPackBSize > 0 for the availability of pre-pack weight.